src/dmd/ctfeexpr.d

```diff
@@ -873,7 +873,9 @@ UnionExp pointerArithmetic(const ref Loc loc, TOK op, Type type, Expression eptr
         error(loc, "CTFE internal error: bad pointer operation");
         goto Lcant;
     }
-    if (indx < 0 || len < indx)
+    // We allow indx to be equal to the len: one position below the right bound.
+    // So we do for the left bound for backward iteration algorithms.
```

Comment:
Reverse iteration algorithms, because the keyword isn't `foreach_backwards`.

Reply:
I don't understand you. What is the problem with `foreach_backwards`? It isn't related to the issue.

Reply:
It triggers semantic warnings for native English speakers. ;-)

Reply:
Got it:) thanks